---
title: C6320
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C6320
helpviewer_keywords:
- C6320
ms.assetid: fb9e568e-b3d4-4ce2-a276-a64ad74d7b1e
ms.openlocfilehash: 979143e9bf95f0692c63cd68a8f0747c67a82094
ms.sourcegitcommit: 7bea0420d0e476287641edeb33a9d5689a98cb98
ms.translationtype: MT
ms.contentlocale: fr-FR
ms.lasthandoff: 02/17/2020
ms.locfileid: "77419697"
---
# <a name="c6320"></a>C6320

> AVERTISSEMENT C6320 : l’expression de filtre d’exception est la constante EXCEPTION_EXECUTE_HANDLER. Cela peut masquer les exceptions qui n’étaient pas destinées à être gérées

Cet avertissement indique l’effet secondaire de l’utilisation de EXCEPTION_EXECUTE_HANDLER constante dans __except bloc. Dans ce cas, l’instruction dans le bloc \__except s’exécute toujours pour gérer l’exception, y compris les exceptions que vous ne souhaitez pas gérer dans une fonction particulière. Il est recommandé de vérifier l’exception avant de la gérer.

## <a name="example"></a>Exemple

Le code suivant génère cet avertissement, car le bloc __except essaiera de gérer les exceptions de tous les types :

```cpp
#include <stdio.h>
#include <excpt.h>

void f(int *p)
{
   __try
   {
      puts("in try");
      *p = 13;  // might cause access violation exception
      // code ...
   }
   __except(EXCEPTION_EXECUTE_HANDLER) // warning
   {
      puts("in except");
      // code ...
   }
}
```

Pour corriger cet avertissement, utilisez `GetExceptionCode` pour rechercher une exception particulière avant de la gérer, comme indiqué dans le code suivant :

```cpp
#include <stdio.h>
#include <windows.h>
#include <excpt.h>

void f(int *p)
{
   __try
   {
      puts("in try");
      *p = 13;    // might cause access violation exception
      // code ...
   }
   __except(GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)
   {
      puts("in except");
      // code ...
   }
}
```

## <a name="see-also"></a>Voir aussi

[try-except, instruction](/cpp/cpp/try-except-statement)
